chef/cookbooks/provisioner/templates/suse/crowbar_register.erb

```diff
@@ -152,19 +152,21 @@ if test "x$MD5_ADMIN" != "x$MD5_LOCAL"; then
     exit 1
 fi
 
+#we need to know an architecture we are running on
+ARCH=`uname -m`
 
 # Setup the repos
 # ---------------
 
 <% @repos.keys.sort.each do |name| %>
-zypper -n ar "<%= @repos[name][:url] %>" "<%= name %>"
+zypper -n ar "<%= @repos[name][:url] %>" "<%= name %>-$ARCH"
 <% end %>
 
 # Cloud-PTF has an unknown key, and this is expected
 <% if @target_platform_version.to_f < 12.0 -%>
-zypper -n --gpg-auto-import-keys refresh -r SLE-Cloud-PTF
+zypper -n --gpg-auto-import-keys refresh -r SLE-Cloud-PTF-$ARCH
 <% else -%>
-zypper -n --gpg-auto-import-keys refresh -r SLE12-Cloud-Compute-PTF
+zypper -n --gpg-auto-import-keys refresh -r SLE12-Cloud-Compute-PTF-$ARCH
```

Comment:
Yes, qa_crowbarsetup needs to adapt to that change. I could easily make change to biosdevname only.